packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -676,8 +676,309 @@ void main() {
+          await gesture.moveBy(Offset(0, -scrollExtent));
+
+          await tester.pump();
+
+          expect(
+            globalKey.currentState.innerController.position.pixels +
+                globalKey.currentState.outerController.position.pixels,
+            scrollExtent,
+          );
+
+          // The outer scroll view cannot scroll past its height, which should be kToolbarHeight as a SliverAppBar represents the outer body.
+          final double expectedOuterControllerExtent = scrollExtent > kToolbarHeight ? kToolbarHeight : scrollExtent;
+          expect(
+            globalKey.currentState.outerController.position.pixels,
+            expectedOuterControllerExtent,
+          );
+
+          // The inner controller should only start scrolling once the outer controller has been scrolled to its full extent.
+          final double expectedInnerControllerExtent = scrollExtent < kToolbarHeight ? 0 : scrollExtent - kToolbarHeight;
+          expect(
+            globalKey.currentState.innerController.position.pixels,
+            expectedInnerControllerExtent,
+          );
+        });
+  }
+
+  testWidgets('NestedScrollViewState.outerController should correspond to NestedScrollView.controller', (WidgetTester tester) async {
+    final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+    final ScrollController scrollController = ScrollController();
+
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: Localizations(
+        locale: const Locale('en', 'US'),
+        delegates: const <LocalizationsDelegate<dynamic>>[
+          DefaultMaterialLocalizations.delegate,
+          DefaultWidgetsLocalizations.delegate,
+        ],
+        child: MediaQuery(
+          data: const MediaQueryData(),
+          child: NestedScrollView(
+            controller: scrollController,
+            key: globalKey,
+            body: ListView.builder(
+              itemBuilder: (BuildContext context, int index) {
+                return Text('$index');
+              },
+            ),
+            headerSliverBuilder: (_, __) => <Widget>[const SliverAppBar()],
+          ),
+        ),
+      ),
+    ));
+
+    // The scrollController should be equal to the outerController defined in the state.
+    expect(
+      scrollController.position,
+      globalKey.currentState.outerController.position,
+    );
+  });
+
+  group('NestedScrollViewState controllers interactions', () {
+    testWidgets('inner body not scrolled, header fully expanded', (WidgetTester tester) async {
+      final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+
+      await tester.pumpWidget(NestedScrollViewWithGlobalKey(globalKey: globalKey));
+
+      expect(globalKey.currentState.outerController.position.pixels, 0);
+      expect(globalKey.currentState.innerController.position.pixels, 0);
+
+      const double innerPosition = 0e0, outerPosition = kToolbarHeight;
```

Comment:
Can you just write the numbers out instead of using the e notation? The numbers are not that big.